### ISSUE TITLE
Printing Issues

### ISSUE THREAD

**Issue by brettemoore** [matt-dale/designdbonline#45]
Matt, 
Thanks for taking on this project. I really like the database functions so far. I'm having trouble getting printing working correctly so far. I've tried several different combinations of computers, browsers, and printers, but nothing will line up quite right. Same goes with the PDF version. The labels look great, but I can't make the spacing work. Any thoughts on "best use" printer, OS, browser etc. I'm Mac based at work, and have both at home. At work all the printers are HP's. I haven't tried to print at home yet, so I'll give that a shot tonight. My home printer is a Canon.

For clarification; tried printing from OSX 10.11.5 to HP Laserjet 1022 from Safari and Chrome.
Printed from OSX 10.11.6 to HP Laserjet 1022, HP Pro400 from Safari and Chrome.
Printed from Windows 10 to HP Pro400 from Internet Explorer.

**Comment by matt-dale** [comment]
I agree that this can be quite frustrating.  Once you find the right settings it should be better...
If you are having issues printing in the browser, the best way I found is to print to PDF, then open in Adobe Reader and print from there.  Use the "Actual Size" option (not fit to page).  Printers should not matter when printing PDFs.  Working with Adobe should get you going.  Let me know if this works.

**Comment by brettemoore** [comment]
Some improvement in printing from PDF but still not quite there. 
If I use the "Print as PDF" button, the print I get is about 3/16-1/4 of an inch too high. The spacing seems fine, and horizontally everything is good. 

If I print to PDF from the "Print Selected Cables and Breakouts" button, the resulting print is 1/16th too low, but the vertical spacing between cells is off, it creeps up as you move down the page. 

 I'm using Avery 5167 labels. I haven't done as much experimenting on the 5160's yet.

**Comment by matt-dale** [comment]
Hmm...
When you print using the "or Print as PDF" button, are you saving the resulting output, then opening in Adobe Reader?  I have found that the in-browser PDF rendering causes issues with the layouts, but Adobe handles it correctly, but not inside the browser. 

As far as the normal "Print Selected Cables and Breakouts" button, this is rendered with the browser's CSS rendering engine.  I have the best results using Chrome on Windows 10, and when printing, click the Plus for "More Settings"  and set the margins to None.  This print method is not too great... It generally works, but sometimes I have the same issue that you do with the 1/6" creep which seems to be caused by the browser....

**Comment by brettemoore** [comment]
I tried a couple different Print as PDF options, but there didn't seem to be a good way to save the output file directly from the browser. I saved the PDF and then opened with Adobe to print, but it probably wasn't completely absent of the browser rendering the PDF. I'll try at home tonight on my Windows machine and see if that helps things line up.

With Custom Margins, I can get it close printing the PDF from Chrome in OS X, if I could do custom margins on the PDF from Adobe we'd be all set.
